### DISCUSSION TITLE
Bots assigned to deconstruct seemingly do not need to be near a deconstructor to turn it on

### DISCUSSION THREAD

**Discussion by ghost** [FakeFishGames/Barotrauma#13918] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Bots assigned to deconstruct items in a deconstructor do not need to be near the deconstructor to turn it on.
![image](https://github.com/FakeFishGames/Barotrauma/assets/86113712/b9633a1b-e7e5-4e78-8297-07fae9472f23)
![image](https://github.com/FakeFishGames/Barotrauma/assets/86113712/050d82b0-b5bf-4f44-93df-6022203dbd85)
![image](https://github.com/FakeFishGames/Barotrauma/assets/86113712/09608459-2ddb-4a3a-ae3c-73916fe14d5b)

### Reproduction steps

1. Assign a bot to deconstruct an item that is already inside a deconstructor while they are not near it.

(easily achieved by turning off power, assigning them to deconstruct an item inside a deconstructor, giving them a wait order far from the deconstructor and then dismissing the wait order. Other methods like simply teleporting them also work)

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.5.0 (Blood in the Water Update, hotfix 1)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/pull/5348/commits/f98cec4cfc8f36f7c50bde3893da6ceade9d20e0

**Comment by Jasontti** [comment]
Tested the fix and no issues found.
Closing discussion as resolved

**Comment by Regalis11** [comment]
Worth noting that the fix did not make it in time for the next (June) update, but will be included in a future patch/update.
